tuning/utils/collators.py

```diff
@@ -11,45 +11,85 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-# Third Party
-import torch
+
+# Local
+from tuning.utils.utils import try_convert_bytes_dict_to_pil
 
 
 class VisionDataCollator:
+    """
+    A data collator specialized for multi-modal (text + image) inputs.
+    It uses a processor (e.g., LlavaProcessor or MllamaProcessor) to
+    combine text and images into model-ready tensors.
+
+    For padding-free tuning, configure the processor's arguments
+    in `processor_kwargs`, for example:
+        processor_kwargs = {
+            "padding": False,
+            "max_length": 1024,
```

Comment:
It this something the user can pass in via JSON as well? Would the user need to set padding_free as a flag and processor_kwargs?